spec/runner_spec.rb

```diff
@@ -0,0 +1,185 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'validate_environment_varables!' do
+  let(:envvars) { Jacaranda::Runner.required_environment_variables }
+  subject { -> { Jacaranda::Runner.validate_environment_variables! } }
+
+  context 'when all environment variables are set' do
+    before(:each) { envvars.each { |var| set_environment_variable(var, Faker::Name.first_name) } }
+
+    it 'does not exit' do
+      is_expected.to_not raise_error
+    end
+  end
+
+  context 'when any environment variables are not set' do
+    before(:each) { envvars.each { |var| unset_environment_variable(var) } }
+
+    it 'exits' do
+      is_expected.to raise_error(SystemExit)
+    end
+  end
+
+  after(:each) { restore_env }
+end
+
+describe 'Jacaranda#run' do
+  let(:names) { Array.new(3) { Faker::Name.first_name } }
+  let(:runners) do
+    sorted = names.sort_by { |c| c.to_s.split('::').last }
+    sorted.map { |name| Object.const_set(name, Class.new(Jacaranda::Runner)) }
+  end
```

Comment:
That makes sense @auxesis 👍 Sounds useful. I think this solution smells a little bit of the (Mystery Guest)[https://robots.thoughtbot.com/mystery-guest] but because this is still a small project I recon that's fine 👍 Go forth!